SourceX/miniwin/misc_io.cpp

```diff
@@ -33,11 +33,13 @@ HANDLE CreateFileA(LPCSTR lpFileName, DWORD dwDesiredAccess, DWORD dwShareMode,
 	file->path = name;
 	if (dwCreationDisposition == DVL_OPEN_EXISTING) {
 		// read contents of existing file into buffer
-		std::ifstream filestream(file->path, std::ios::binary);
-		if (!filestream.fail()) {
-			file->buf.insert(file->buf.begin(),
-			    std::istreambuf_iterator<char>(filestream),
-			    std::istreambuf_iterator<char>());
+		std::ifstream filestream(file->path, std::ios::in | std::ios::binary);
+		if (filestream) {
+			filestream.seekg(0, std::ios::end);
+			file->buf.resize(filestream.tellg());
+			filestream.seekg(0, std::ios::beg);

```

Comment:
Not 100% sure, but I think can just be `filestream.seekg(0)`
http://www.cplusplus.com/reference/istream/istream/seekg/